lib/doctrine.js

```diff
@@ -324,6 +324,10 @@
                     name += scanIdentifier(last);
 
                 }
+                if(source.charCodeAt(index) === 0x5B  /* '[' */ && source.charCodeAt(index + 1) === 0x5D  /* ']' */){
+                    name += advance();
+                    name += advance();
+                }
                 while (source.charCodeAt(index) === 0x2E  /* '.' */ ||
```

Comment:
This loop seems to handle nesting `MemberExpression`.
So I think the code for `[]` should be inside of this loop.